### PR TITLE
[otp] Minor width mismatch fixes when computing parameters

### PR DESCRIPTION
**Pull request by rswarbrick**
These cause Verilator lint errors. The second patch in the PR is pretty trivial, but there's a bit of thought that went into the first one: I think it fixes a bug that was present in the previous code if size + offset is large.

### CODE REVIEW

File: hw/ip/otp_ctrl/rtl/otp_ctrl_dai.sv [hunk 1]
```diff
@@ -637,15 +637,23 @@ module otp_ctrl_dai
 
   logic [NumPart-1:0] part_sel_oh;
   for (genvar k = 0; k < NumPart; k++) begin : gen_part_sel
-    localparam logic [OtpByteAddrWidth:0] PartEnd = (OtpByteAddrWidth+1)'(PartInfo[k].offset) +
-                                                    (OtpByteAddrWidth+1)'(PartInfo[k].size);
+    localparam integer PartEndInt = int'(PartInfo[k].offset) + int'(PartInfo[k].size);
+    localparam integer DigestOffsetInt = PartEndInt - ScrmblBlockWidth / 8;
+
+    // PartEnd has an extra bit to cope with the case where offset + size is more than (1 <<
+    // OtpByteAddrWidth). We keep the same extra bit for DigestOffset, because subtracting
```
Comment (msfschaffner) [inline hw/ip/otp_ctrl/rtl/otp_ctrl_dai.sv:644]: the memory map generation script ensures that this cannot happen.
i.e., offset + size can in the worst case be exactly `1<<OtpByteAddrWidth`, in which case this extra bit is needed.
so any nonzero number will be enough to clear it (and the scrambling block size is always a multiple of bytes).

how about rewording the comment a bit to reflect this, and maybe add a static assertion that checks that PartEnd can never be larger than `1<<OtpByteAddrWidth`?
the rest of the change itself is fine, especially if it helps to clean up a couple of lint messages.

File: hw/ip/otp_ctrl/rtl/otp_ctrl_dai.sv [hunk 2]
```diff
@@ -637,15 +637,23 @@ module otp_ctrl_dai
...
+
     assign part_sel_oh[k] = (dai_addr_i >= PartInfo[k].offset) & ({1'b0, dai_addr_i} < PartEnd);
-    // Needed for other address and access checks.
-    localparam logic [OtpByteAddrWidth-1:0] DigestOffset = OtpByteAddrWidth'(PartEnd -
-                                                                             ScrmblBlockWidth/8);
-    assign digest_addr_lut[k] = DigestOffset >> OtpAddrShift;
+    assign digest_addr_lut[k] = DigestAddrLut[OtpByteAddrWidth-1:0];
   end
 
+  `ASSERT_INIT(PositiveOtpAddrShift_A, OtpAddrShift > 0)
```
Comment (msfschaffner) [inline hw/ip/otp_ctrl/rtl/otp_ctrl_dai.sv:656]: I have tried to keep this mostly agnostic to the actual interface size (8 or 16 bits in this case) where possible.
I cannot claim that this works out of the box for 8bit, but in that case this assertion would not be true anymore.

Anyway, for the same reason mentioned above I believe this assertion is not needed, but I would replace it with the `PartEnd <= 1<<OtpByteAddrWidth` assertion.

**Review by msfschaffner — COMMENTED** [review]
Thanks for the cleanup @rswarbrick.
See comments inline.

File: hw/ip/otp_ctrl/rtl/otp_ctrl_dai.sv (hunk 1 above)
Comment (rswarbrick) [reply hw/ip/otp_ctrl/rtl/otp_ctrl_dai.sv:644]: Ahah, yep that makes sense. I'll re-do things to use that assertion instead (and rewrite the comment accordingly)

**Review by msfschaffner — APPROVED** [review]
LGTM, thanks!
